src/common/libutil/fluid.c

```diff
@@ -181,7 +342,10 @@ int fluid_decode (const char *s, fluid_t *fluidp, fluid_string_type_t type)
             if (rc != 8)
                 return -1;
             break;
-
+        case FLUID_STRING_F58:
+            if (fluid_f58_decode (&fluid, s) < 0)
+                return -1;
+            break;
         default:
             return -1;
```

Comment:
not from this PR, but since you're here, this needs errno set, and also the `mn_decode() != 8` case (my bad)